### ISSUE TITLE
Module doesnt honor specified GKE master version

### ISSUE THREAD

**Issue by kkapoor1987** [terraform-google-modules/terraform-google-kubernetes-engine#1460]
### TL;DR

When both the release channel and master version are specified the module ignores the master version which creates a discrepancy in the intended cluster version vs what the module is creating. 

https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/private-cluster/cluster.tf#L57

### Expected behavior

If you user specifies the version module should either use that version or fail provisioning with error message version not supported

### Observed behavior

We use this module to deploy GKE clusters. We use a Stable release channel but control the master version so that we can reliably deploy the same K8s version across our fleet. 

### Terraform Configuration

```hcl
module "gke" {
  source             = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
  version            = "23.3.0"
  project_id         = var.AccountID
  name               = var.ClusterName
  kubernetes_version = local.latest_gke_patched_version
  regional           = true
  region             = var.Region
  network            = var.ClusterName
  subnetwork         = var.ServiceSubnets[0]
  ip_range_pods      = var.PodRange
  ip_range_services  = var.ClusterServiceRangeName
  horizontal_pod_autoscaling = true
  database_encryption        = local.cluster_encryption_config
  enable_private_endpoint  = false
  enable_private_nodes     = true
  remove_default_node_pool = true
  network_policy           = true
  monitoring_service       = "none"
  master_ipv4_cidr_block   = "10.0.64.0/28"
  logging_service = "logging.googleapis.com/kubernetes"
  cluster_resource_labels           = local.tags
  node_pools                 = local.node_pools
  master_authorized_networks = local.master_authorized_networks
  
  # Release Channel configuration
  release_channel = var.release_channel_version
  maintenance_exclusions = [{
    name = "minor_upgrade_exclusion"
    start_time = timestamp()
    end_time = timeadd(timestamp(),local.end_time_in_hrs)
    exclusion_scope = "NO_MINOR_UPGRADES" # Scope should be one of: NO_UPGRADES | NO_MINOR_UPGRADES | NO_MINOR_OR_NODE_UPGRADES
  }]
}
```

### Terraform Version

```sh
1.0.11
```

**Comment by slimatic** [comment]
Can this be reopened?
